src/Modules/AuthModule/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,6 +1,251 @@
 
 export default function ChangePassword() {
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm<FormDataChangPass>();
+
+  const { baseUrl } = useContext(AuthContext);
+
+  // Password eye
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  const togglePasswordVisibility = (): void => {
+    setShowPassword((prevState: boolean) => !prevState);
+  };
+
+  const [showNewPassword, setShowNewPassword] = useState<boolean>(false);
+
+  const togglePasswordVisibility2 = (): void => {
+    setShowNewPassword((prevState: boolean) => !prevState);
+  };
+
+  const [showConfirmPassword, setShowConfirmPassword] = useState<boolean>(false);
+
+  const togglePasswordVisibility3 = (): void => {
+    setShowConfirmPassword((prevState: boolean) => !prevState);
+  };
+
+  const onSubmit = async (data: FormDataChangPass) => {
+    try {
+      let response = await axios.put(`${baseUrl}/Users/ChangePassword`,
+      data,
+      {
+        headers: { Authorization: `Bearer ${ localStorage.getItem("token")}` },
+      }
+    )
+      // logout();
+      toast.success('Password has been updated successfully');
+    }
+    catch(error: any) {
+      toast.error(error.response.data.message);
+    }
+  };
+
   return (
-    <div>ChangePassword</div>
+    <>
+      <div className="background-pass">
+        <div className="mt-1 text-center">
+                <img src={logo} alt="" className="w-50" />
+        </div>
+        <div className="form">
+          <form
+            className="p-3 rounded-4 my-5"
+            onSubmit={handleSubmit(onSubmit)}
+          >
+            <div className="text my-1">
+              <h2>Change Password</h2>
+            </div>
+                <div className="row">
+                  <div className="col-md-11 mt-3 p-3 ">
+                    {/* Old Password */}
+                    <div className="visibilty-password input m-3">
+                      <label
+                        className="color-text "
+                        htmlFor="exampleFormControlInput2"
+                      >
+                        oldPassword
+                      </label>
+                      <input
+                        type={showPassword ? "text" : "password"}
+                        className="form-control shadow-none"
+                        style={{
+                          backgroundColor: showPassword
+                            ? "transparent"
+                            : "#fff",
+                          borderTop: showPassword ? "none" : "",
+                          borderRight: showPassword ? "none" : "",
+                          borderLeft: showPassword ? "none" : "",
+                          borderRadius: showPassword ? "0px" : "",
+                          color: showPassword  ? "white" :  "",
+                        }}
+                        placeholder="Enter your Old Password"
+                        {...register("oldPassword", {
```

Comment:
Email, password and other validation will be used in multiple places. I consider making a file containing all needed validations and then use this validation in our application. This will make our code more concise.

You can look at this example:
https://github.com/AnasKhaled7/PMS-React/blob/main/src/lib/InputValidator.ts